File: Client/Protocol/Message.cpp
//
//  Created by Stasel
//

#include "Message.h"

Message::Message(const UUID& senderClientID, UInt32 messageID, MessageType type, const Bytes& content)
    : senderClientID(senderClientID), messageID(messageID), type(type), content(content) { }

const UUID& Message::getSenderClientID() const {
    return senderClientID;
}
const UInt32 Message::getMessageID() const {
    return  messageID;
}
const MessageType Message:: getType() const {
    return type;
}
const Bytes& Message::getContent() const {
    return content;
}

File: Server/Message.py
import uuid


class Message:
    def __init__(self, messageId: int, toClient: uuid.UUID, fromClient: uuid.UUID, messageType: int, content: bytes):
        assert isinstance(messageId, int)
        assert isinstance(toClient, uuid.UUID)
        assert isinstance(fromClient, uuid.UUID)
        assert isinstance(messageType, int)
        assert isinstance(content, bytes)

        self.id = messageId
        self.toClient = toClient
        self.fromClient = fromClient
        self.type = messageType
        self.content = content
